lib/Pothole/Pothole.php

```
@@ -102,15 +98,13 @@ private function _validate($details)
         if (filter_var($details['bad'], FILTER_VALIDATE_INT, $options)) {
             $this->_columns['rating'] = $details['bad'];
         } else {
-            $this->_error .= "Please rate the pothole/n";
-            $result = false;
+            throw new Exception("Invalid pothole rating");
         }
 
         if (isset($details['report-nick']) && $details['report-nick']) {
             $this->_columns['nickname'] = $purifier->purify($details['report-nick']);
         } else {
-            $this->_error .= "Please enter a nickname/n";
-            $result = false;
+            throw new Exception("Please enter a nickname");
```

For these - since multiple things can be wrong I'd rather throw a single exception listing everything rather than having the user resubmit and get the next error.